generals/gui/event_handler.py

```diff
@@ -1,77 +1,179 @@
 import pygame
+from pygame.event import Event
 
 from .properties import Properties
+from generals.core import config as c
+
+######################
+# Replay keybindings #
+######################
+RIGHT = pygame.K_RIGHT
+LEFT = pygame.K_LEFT
+SPACE = pygame.K_SPACE
+Q = pygame.K_q
+R = pygame.K_r
+H = pygame.K_h
+L = pygame.K_l
+
+
+class Command:
+    def __init__(self):
+        self.quit: bool = False
+
+
+class ReplayCommand(Command):
+    def __init__(self):
+        super().__init__()
+        self.frame_change: int = 0
+        self.speed_change: float = 1.0
+        self.restart: bool = False
+        self.pause: bool = False
+
+
+class GameCommand(Command):
+    def __init__(self):
+        super().__init__()
+        raise NotImplementedError
+
+
+class TrainCommand(Command):
+    def __init__(self):
+        super().__init__()
 
 
 class EventHandler:
-    def __init__(self, properties: Properties, from_replay=False):
+    def __init__(self, properties: Properties):
         """
         Initialize the event handler.
 
         Args:
             properties: the Properties object
-            from_replay: bool, whether the game is from a replay
         """
         self.properties = properties
-        self.from_replay = from_replay
+        self.mode = properties.mode
+        self.handler_fn = self.initialize_handler_fn()
+        self.command = self.initialize_command()
+
+    def initialize_handler_fn(self):
```

Comment:
Consider strategy pattern:
classes `GameEventHandler`, `ReplayEventHandler`, `TrainEventHandler`
implementing interface `AbstractEventHandler`
which implement `handle_key_controls`, `handle_mouse_click` and `command`

This way, you can save the correct instance in `self.handler` and always use the same method calls.